io.github.lpub3d.LPub3D.json

```diff
@@ -0,0 +1,183 @@
+{
+  "id": "io.github.lpub3d.LPub3D",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15",
+  "sdk": "org.kde.Sdk",
+  "command": "lpub3d24",
+  "rename-icon": "lpub3d",
+  "rename-desktop-file": "lpub3d.desktop",
+  "rename-appdata-file": "io.github.lpub3d.appdata.xml",
+  "build-options": {
+    "env": {
+      "LP3D_3RD_PARTY_DIR": "/stage/lpub3d_linux_3rdparty",
+      "LP3D_DIST_DIR_PATH": "/stage/lpub3d_linux_3rdparty",
+      "LDRAWDIR": "/stage/lpub3d_linux_3rdparty/ldraw",
+      "PLATFORM_CODE": "osl",
+      "TARGET_CPU": "$(uname -m)",
+      "PLATFORM_VER": "$(. /etc/os-release 2>/dev/null; [ -n \"$VERSION_ID\" ] && echo $VERSION_ID || true)"
+    }
+  },
+  "cleanup": [
+    "/stage"
+  ],
+  "finish-args": [
+    "--device=dri",
+    "--share=ipc",
+    "--socket=x11",
+    "--filesystem=home",
+    "--env=TMPDIR=/var/tmp",
+    "--env=QT_ENABLE_HIGHDPI_SCALING=1"
+  ],
+  "modules": [
+    {
+      "name": "part-archives",
+      "buildsystem": "simple",
+      "build-commands": [
+        "unzip -d ${LP3D_3RD_PARTY_DIR} -q /stage/lpub3d_linux_3rdparty/complete.zip",
+        "if test -d ${LP3D_3RD_PARTY_DIR}/ldraw; then echo 'LDraw library extracted.'; fi"
+      ],
+      "sources": [
+        {
+          "type": "file",
+          "dest-filename": "complete.zip",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/complete-20-03.zip",
+          "sha256": "85cf747b5923f18a8d0f37d0cef647636d70ddfeda5de49ea019e7e440f9e4e0"
+        },
+        {
+          "type": "file",
+          "dest-filename": "lpub3dldrawunf.zip",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/lpub3dldrawunf-01-04-21.zip",
+          "sha256": "f0266006c4b7114e4f4719801696fbe8780cf9d767e948fb6478c752989cbee8"
+        },
+        {
+          "type": "file",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/tenteparts.zip",
+          "sha256": "7e1a68490b1be11bddcfb3180151b42ffbb85ce3aa8ec3f7431bee5c04cc1ac5"
+        },
+        {
+          "type": "file",
+          "url": "https://github.com/trevorsandy/lpub3d_libs/releases/download/v1.0.1/vexiqparts.zip",
+          "sha256": "e2f4114c8ff6bfe7e5a43b6c88fb47f32328ddb147578456ec0825aa76803b20"
+        }
+      ],
+      "post-install": [
+        "mv -f complete-20-03.zip ${LP3D_3RD_PARTY_DIR}/complete.zip",      
+        "mv -f lpub3dldrawunf-01-04-21.zip ${LP3D_3RD_PARTY_DIR}/lpub3dldrawunf.zip",
+        "mv -f tenteparts.zip ${LP3D_3RD_PARTY_DIR}/",
+        "mv -f vexiqparts.zip ${LP3D_3RD_PARTY_DIR}/"
+      ]
+    },
+    {
+      "name": "ldglite",
+      "buildsystem": "qmake",
+      "config-opts": [
+        "CONFIG+=3RD_PARTY_INSTALL=/stage/lpub3d_linux_3rdparty/ldglite_renderer",
+        "CONFIG+=release",
+        "CONFIG+=BUILD_CHECK"
+      ],
+      "sources": [
```

Comment:
ldglite require `freeglut` which require `glu`.

Reply:
So is “bump your nose” the only approach to determine missing dependencies ? I guess I could scour the other repos looking for packages that use these libs.

Cheers,

Reply:
One I thing I do before submitting a flatpak is building locally using `flatpak-builder`. I admit figuring out what needs to be build is sometime a guess job. Just trying to help mentioning what is needed here.

Currently (with my local changes) I'm stopped at the requirement to have `osmesa`. Mesa is in the runtime, but it's build without OSMesa. And the only other flatpak I could find, is Octave which removed the need for OSMesa a while back.

Reply:
I did look briefly at setting up a local flatpack-builder. But I learned I needed to set up flatpack for that so I stopped there - it difficult enough setting up a distro much less the whole platform. Coming off my experience of setting up AppImage and Open Build Service, I guess my expectations were off with regards to the difference in approach between flatpack and those platforms. 

I also managed to collect several example yml files to round out the dependency requirements. I haven’t found one for OSMesa yet either. I’ll keep looking though.

Cheers,

Reply:
There is not much setup to insall flatpak-builder and have it run IMHO. The biggest thing is probably intaling the SDK which can take some time depending on the speed of your internet connection. And the few GB of disk space.

For OSmesa I have this, but it probably need refining:
```json
 {
          "name": "mesa",
          "buildsystem": "meson",
          "config-opts": [
            "-Dosmesa=classic",
            "-Ddri-drivers=[]",
            "-Dplatforms=['x11']",
            "-Dgallium-drivers=[]",
            "-Dvulkan-drivers=[]"
          ],
          "sources": [
            {
              "type": "archive",
              "url": "https://archive.mesa3d.org//mesa-20.2.6.tar.xz",
              "sha256": "f12ca3c6c622f11cd79ad66b4220f04514fa96f795062fe92a37339ab19885db"
            }
          ]
        }
```
And you need to set the following for ldglite in the `config-opts`
```json
        "QMAKE_LIBDIR=/app/lib",
```

Reply:
By refining, I mean cleaning up unused stuff.

Reply:
I have published by WiP based on your PR https://github.com/hfiguiere/flathub/tree/io.github.lpub3d.LPub3D 

It still fail to build.

Reply:
Excellent. The OSMesa bit is a good starting point, I think only the build flags might need to be updated. I’ll take a look at what I use between LDView and LDGLite.

> There is not much setup to insall flatpak-builder and have it run IMHO. 

Do you know if it will work on Microsoft’s WLS (Ubuntu 16.04) ? Will configure without arguments be ok because none are suggested ?

Cheers,

Reply:
Five year old Ubuntu, that's pushing the difficulty level to the max...